src/main/java/dao/impl/CartDaoImpl.java

```diff
@@ -0,0 +1,89 @@
+package dao.impl;
+
+import dao.CartDao;
+import model.Cart;
+import model.Product;
+import model.User;
+import org.apache.log4j.Logger;
+import util.DBConnection;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Optional;
+
+public class CartDaoImpl implements CartDao {
+
+    private static final Logger logger = Logger.getLogger(UserDaoImpl.class);
+
+    @Override
+    public void addCart(Cart value) {
+
+        String sql = String.format("INSERT INTO cart(idUser) VALUES('%s')",
+                                    value.getUser().getUserID());
+        try (Connection connection = DBConnection.getConnection();
+             Statement statement = connection.createStatement()) {
+            statement.execute(sql);
+            logger.info(value + " added to db");
+        } catch (SQLException e) {
+            logger.error("SQl exception " + e);
+        }
+    }
+
+    @Override
+    public Optional<Cart> getCart(User value) {
+
+        Optional<Cart> cart = Optional.empty();
+        ArrayList<Product> products = new ArrayList<>();
+        String sql = "SELECT idUser, cart.idCart FROM cart " +
```

Comment:
maybe we can use one sql query in this method?

Reply:
> maybe we can use one sql query in this method?

fixed :)